stbt_rig.py

```diff
@@ -554,40 +554,34 @@ def cmd_setup(args, node_id):
             v = "3.10"
         else:
             v = "3.6"
-        python_executable_names = (["python%s" % v], ["py", "-%s" % v],
-                                   ["python"], ["python3"], ["py", "-3"])
 
-        if not os.path.exists("%s/.venv" % root):
-            python = None
-            for python in python_executable_names:
-                try:
-                    o = to_unicode(subprocess.check_output(
-                        python + ["-c", "import sys; print(sys.version)"],
-                        stdin=open(os.devnull))).strip()
-                    if o.startswith("3."):
-                        if not o.startswith(v + "."):
-                            logger.warning(
-                                "Using Python version %s which may not be "
-                                "fully compatible with stb-tester %s. "
-                                "For best compatibility install Python %s.",
-                                o, stbt_version, v)
-                        break
-                except (subprocess.CalledProcessError, OSError):
-                    # Doesn't exist, or there's something wrong with it
-                    pass
-            else:
-                die("Can't find python %s in PATH" % python_version)
+        system_python_exe, system_python_version = _find_python_executable(
+            v, stbt_version)
 
-            # Create venv
-            subprocess.check_call(python + ['-m', 'venv', '.venv'], cwd=root)
+        venv_dir = os.path.join(root, ".venv")
+
+        if os.path.exists(venv_dir):
+            venv_python_version = _get_python_version([_venv_exe("python", root)])
+            if (system_python_version.startswith(v + ".") and
+                    not venv_python_version.startswith(v + ".")):
+                logging.warning(
+                    "Virtualenv in '%s' has Python version %s; "
+                    "deleting .venv and recreating it with Python %s.",
+                    venv_dir, venv_python_version, v)
+                shutil.rmtree(venv_dir)
```

Comment:
I think it would be good to prompt before running this.  The user might want to back up the `.venv` or record what versions of things they have installed there.

Untested:

```suggestion
                while True:
                    ok = input("Delete and recreate .venv?: Y/N: ").upper().strip()
                    if ok == "Y":
                        shutil.rmtree(venv_dir)
                        break
                    elif ok = "N":
                        break
```

Reply:
Done in c49beb8.

Tested by setting stbt_version to 33, running `./stbt_rig.py setup`, answering `n` and then running it again and answering `y`.